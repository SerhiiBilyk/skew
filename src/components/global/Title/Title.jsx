import React from 'react';
import ReactDOM from 'react-dom';
import styles from './title.scss';
import CSSModules from 'react-css-modules';
import PropTypes from 'prop-types';

class Title extends React.PureComponent {
  constructor(props) {
    super(props);
  }
  render() {
    var {title, sub, theme} = this.props;

    return (
      <div styleName={`title ${this.props.theme}`}>
        <h1>{this.props.title}</h1>
        <p>{this.props.sub}</p>
      </div>
    )
  }
}
Title.propTypes={
theme:PropTypes.oneOf(['white','dark']),
title:PropTypes.string.isRequired
}
export default CSSModules(Title, styles, {allowMultiple: true});
