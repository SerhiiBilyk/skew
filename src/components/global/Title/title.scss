.title {
  font-family: "Open Sans";
  text-align:center;
  padding:17px 25px 48px;

  h1{
    font-size: 37px;
      line-height:34px;
      margin-bottom:15px;
  }
  p{
    font-size:20px;
      line-height:28px;
    color:  #B9B9B9;
  }
}

.dark {
  color: #2f2f2f;
}
.white{
  color:#ffffff;
}

@include media('zerro','sm-phone'){
  .title{
    padding:17px 0 48px;
    h1{
      font-size:30px;
    }
  }
}
@include media('sm-phone','lg-phone'){
  .title{
    padding:17px 0 48px;
    h1{
      font-size:30px;
    }
  }
}
