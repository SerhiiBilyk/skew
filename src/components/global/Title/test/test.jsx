import React from 'react';
import  {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';
import Title from '../Title.jsx';

// TODO write more tests, it is created only for folder architecture

/**SNAPSHOT*/
it('<Title/> renders correctly dark theme ', () => {
  const tree = renderer.create(
   <Title theme='dark' title='Random'/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
