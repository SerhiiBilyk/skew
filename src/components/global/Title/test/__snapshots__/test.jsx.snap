// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> renders correctly dark theme  1`] = `
<div
  className="title dark"
>
  <h1>
    Random
  </h1>
  <p />
</div>
`;
